JacquardToolkit/JacquardToolkit/JacquardService.swift

```
@@ -126,6 +134,90 @@ public class JacquardService: NSObject, CBCentralManagerDelegate {
         return csvTextStore
     }
     
+    /**
```

For lines 137 to 220, this is definitely be consolidated with parameterization. Something like playTutorial(viewController, gestureName) where first you check if JSConstants.JSURLs.Tutorial contains the passed in gestureName. If not, then fail fast/throw an error. As for the Boolean flags for is_TutorialActivated, this too would potentially be better as something more generalizable/simplified. One idea is to assign a activeTutorial var to the gestureName that’s active and use that for all corresponding Boolean checks.